c_emulator/riscv_sim.c

```diff
@@ -242,12 +249,20 @@ char *process_args(int argc, char **argv)
                     "v::"
                     "l:"
                     "x"
+                    "X"
 #ifdef SAILCOV
                     "c:"
 #endif
-                         , options, NULL);
+                         , options, &option_index);
     if (c == -1) break;
     switch (c) {
+    case 0:     // Not the character '0', but the NULL value.
+      printf ("option %s", options[option_index].name);
+      if (optarg) {
+        printf (" with arg %s", optarg);
```

Comment:
Ditto

Reply:
Also why is this code here?

Reply:
This was added for debug purposes.  I am now using the "only long form options" of getopt.  
Is this useful in the long run?

Reply:
Well, you literally copied it verbatim from glibc's manpage. The whole point of case 0 is to not have to do anything, it does it automatically. The manage only has that example so you can see what's going on. And if you make a typo then you won't hit that case...

So no, it is not useful, it does not belong here, even commented or ifdef-ed out.